docs/performance/sharding.md

```diff
@@ -1,41 +1,133 @@
+
+
+## Sizing considerations
+
+General principles requires careful consideration for cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
 
-If you are looking for an intro to sharding, see {ref}`sharding
-<crate-reference:ddl-sharding>`.
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 3-70 GB of data, with 10-50 GB being the sweet spot for most
+workloads. In large clusters, this often means fewer shards than total CPU cores,
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+Smaller shards also result in reduced Lucene index efficiency, which can adversely
+affect computed search term relevance.
+
+### CPU-to-shard ratio
+
+If most nodes have more shards per table than they have CPUs, the cluster can
+experience performance degradations.
+For example, on clusters with substantial CPU resources (e.g., 8 nodes × 32 CPUs
+= 256 total CPUs), creating 256+ shards per table often proves counterproductive.
+If you don't manually set the number of shards per table, CrateDB will make a
+best guess, based on the assumption that your nodes have two CPUs each.
+
+### 1000 shards per node limit
+
+To avoid _oversharding_, CrateDB by default limits the number of shards per node to
+1_000 as a critical stability boundary. Any operation that would exceed that limit
+leads to an exception.
+For an 8-node cluster, this allows up to 8_000 total shards across all tables.
+Approaching this limit typically indicates a suboptimal sharding strategy rather
+than optimal performance tuning. See also relevant documentation about
+{ref}`table reconfiguration <number-of-shards>` wrt. sharding options.
+
+### Partitions
+
+If you are using {ref}`partitioned tables <crate-reference:partitioned-tables>`,
+note that each partition is clustered into as many shards as you configure
+for the table.
+
+For example, a table with four shards and two partitions will have eight
+shards that can be commonly queried across. But a query that only touches
+one partition will only query across four shards.
+
+How this factors into balancing your shard allocation will depend on the
+types of queries you intend to run.
+
+### Replicas
+
+CrateDB also has replicas of data and this results in additional shards in
+the cluster.
+
+For the purposes of calculating how many shards a table should be clustered
+into, you can typically ignore replica partitions as these are not usually
+queried across for reads.
+
+
+## Caveats
+
+:::{caution}
+:class: hero
+Balancing the number and size of your shards is important for the performance
+and stability of your CrateDB clusters.
 :::
 
-## Optimising for query performance
+### Avoid imbalances
```

Comment:
About _detecting_ imbalances, we will [introduce](https://github.com/crate/cratedb-guide/issues/301#issuecomment-3302735529) the XMover utility at this spot as soon as it is released.

Reply:
Please review, so we can release the utility and mention it here.

- https://github.com/crate/cratedb-toolkit/issues/513#issuecomment-3325642649

Reply:
Tracking this here now.

- GH-377